src/main/java/org/jenkinsci/plugins/docker/traceability/model/jobs/DockerBuildReferenceRun.java

```diff
@@ -0,0 +1,130 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.docker.traceability.model.jobs;
+
+import hudson.model.AbstractBuild;
+import hudson.model.Fingerprint;
+import hudson.model.Result;
+import hudson.model.Run;
+import java.io.File;
+import java.io.IOException;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import jenkins.model.lazy.LazyBuildMixIn;
+
+/**
+ * References Docker items.
+ * This reference allows to workaround {@link Fingerprint} retention policies.
+ * @author Oleg Nenashev
+ */
+public class DockerBuildReferenceRun extends AbstractBuild<DockerBuildReferenceJob, DockerBuildReferenceRun> {
+    
+    private String itemId;
+    private @CheckForNull String itemName;
+    private Type itemType;
+    private long dockerTimestamp;
+    
+    
+    public DockerBuildReferenceRun(DockerBuildReferenceJob job) throws IOException  {
+        super(job);
+    }
+    
+    public DockerBuildReferenceRun(DockerBuildReferenceJob job, File buildDir) throws IOException  {
+        super(job, buildDir);
+    }
+    
+    /**
+     * Sets data items and generates an appropriate description
+     * @param id
+     * @param name
+     * @param type
+     * @param timestamp
+     * @throws IOException Description update error
+     */
+    void set(@Nonnull String id, @CheckForNull String name, @Nonnull Type type, long timestamp)     
+            throws IOException {
+        this.itemId = id;
+        this.itemName = name;
+        this.itemType = type;
+        this.dockerTimestamp = timestamp;
+        setDescription("for "+type.getDisplayName()+" "+ (name != null ? name : id));
+    }
+
+    public String getItemId() {
+        return itemId;
+    }
+
+    public Type getItemType() {
+        return itemType;
+    }
+
+    /**
+     * Optional name of the docker item
+     * @return Name of the docker item
+     */
+    public @CheckForNull String getItemName() {
+        return itemName;
+    }
+ 
+    public long getDockerTimestamp() {
+        return dockerTimestamp;
+    }
+
+    @Override
+    public void run() {
+        onStartBuilding();
+        setResult(Result.SUCCESS);
+        onEndBuilding();
+    }
+
+    /**
+     * Type of the {@link DockerBuildReferenceRun}.
+     */
+    public static enum Type {
+        CONTAINER,
+        IMAGE;
+        
+        /**
+         * Display name of the {@link Type}.
+         * May become localizable later, defaults to {@link #getTypeName()}.
+         */
+        public @Nonnull String getDisplayName() {
+            return getTypeName();
+        }
+        
+        /**
+         * Short unique id of the {@link Type}.
+         */
+        public @Nonnull String getTypeName() {
+            switch(this) {
+                case CONTAINER:
+                    return "container";
+                case IMAGE:
+                    return "image";
+                default:
```

Comment:
shouldn't have defaults in enums - let the compiler and findbugs flag it for you!